modules/pcd/src/lib/parse-pcd.ts

```diff
@@ -312,3 +315,122 @@ function parsePCDBinary(pcdHeader: PCDHeader, data: ArrayBufferLike): HeaderAttr
 
   return {position, normal, color};
 }
+
+/**
+ * @param pcdHeader
+ * @param data
+ * @returns [attributes]
+ */
+function parsePCDBinaryCompressed(PCDheader, data) {
+  const position: number[] = [];
+  const normal: number[] = [];
+  const color: number[] = [];
+
+  const sizes = new Uint32Array(data.slice(PCDheader.headerLen, PCDheader.headerLen + 8));
+  const compressedSize = sizes[0];
+  const decompressedSize = sizes[1];
+  const decompressed = decompressLZF(
+    new Uint8Array(data, PCDheader.headerLen + 8, compressedSize),
+    decompressedSize
+  );
+  const dataview = new DataView(decompressed.buffer);
+
+  const offset = PCDheader.offset;
+
+  for (let i = 0; i < PCDheader.points; i++) {
+    if (offset.x !== undefined) {
+      position.push(
+        dataview.getFloat32(PCDheader.points * offset.x + PCDheader.size[0] * i, LITTLE_ENDIAN)
+      );
+      position.push(
+        dataview.getFloat32(PCDheader.points * offset.y + PCDheader.size[1] * i, LITTLE_ENDIAN)
+      );
+      position.push(
+        dataview.getFloat32(PCDheader.points * offset.z + PCDheader.size[2] * i, LITTLE_ENDIAN)
+      );
+    }
+
+    if (offset.rgb !== undefined) {
+      color.push(
+        dataview.getUint8(PCDheader.points * offset.rgb + PCDheader.size[3] * i + 0) / 255.0
+      );
+      color.push(
+        dataview.getUint8(PCDheader.points * offset.rgb + PCDheader.size[3] * i + 1) / 255.0
+      );
+      color.push(
+        dataview.getUint8(PCDheader.points * offset.rgb + PCDheader.size[3] * i + 2) / 255.0
+      );
+    }
+
+    if (offset.normal_x !== undefined) {
+      normal.push(
+        dataview.getFloat32(
+          PCDheader.points * offset.normal_x + PCDheader.size[4] * i,
+          LITTLE_ENDIAN
+        )
+      );
+      normal.push(
+        dataview.getFloat32(
+          PCDheader.points * offset.normal_y + PCDheader.size[5] * i,
+          LITTLE_ENDIAN
+        )
+      );
+      normal.push(
+        dataview.getFloat32(
+          PCDheader.points * offset.normal_z + PCDheader.size[6] * i,
+          LITTLE_ENDIAN
+        )
+      );
+    }
+  }
+
+  return {
+    position,
+    normal,
+    color
+  };
+}
+
+function decompressLZF(inData, outLength) {
```

Comment:
Please 
- add JSDoc header describing what the function does (link to format spec if possible)
- add typescript types.

Also I assume this is copied from some existing repo, if so can we add a link and mention license and provide attribution? Possibly put this util in a separate file?

Reply:
added docs and types please review it  again

Reply:
> Please
> 
> * add JSDoc header describing what the function does (link to format spec if possible)
> * add typescript types.
> 
> Also I assume this is copied from some existing repo, if so can we add a link and mention license and provide attribution? Possibly put this util in a separate file?

added docs and types please review it again.